### DISCUSSION TITLE
add a separator feature to group habits

### DISCUSSION THREAD

**Discussion by LennTea** [iSoron/uhabits#1281] [Feature Requests]
Hello!

I was thinking that it would be useful to have the possibility of adding a separator (possibly just a hr line or something) to separate groups of habits. 

So far I move them around to sort them and add an emoji at the beginning of the line for "category", but it makes things a bit busy on the screen. It would be easier if I could just add a separator between groups of lines/habits, thanks

**Comment by hiqua** [comment]
How would that work with sorting?

**Reply by LennTea** [reply]
We can move habits by long press-drag. But that's still a potentially long list in one block. I have habits related to various topics. I pressed-dragged the habits to put together under each other, stuff that's related, but there's no separation with the next habits that are about something else. So I thought just even a line separator would help to make things more readable.

Ex instead of

Habit 1 topic A
H 2 t A
H 3 t A
H4 t A
H 5 t B
H 6 t B
H 7 t C

It would be 
H 1 t A
H 2 t A
H 3 t A
H 4 t A
___
H 5 t B
H 6 t B
___
H 7 t B

For example

**Reply by hiqua** [reply]
Sorry I meant: what should happen to the separators if the user sorts by name or by color etc.?

**Reply by decadent** [reply]
Separators mostly make sense only with manual sorting, so the simplest approach is to not display them when a different order is selected. I hope that Loop restores the user-defined order when switching back to manual sorting, and separators could thus also be shown again. (However I'm afraid to test this assumption.)

Separators *could* have meaning when the habits are ordered by the color—serving as a second method of grouping, on top or parallel to the colors. So conceivably in this mode the app could remember which separator is placed next to which entry, and keep the same arrangement when color-sorting is used again. This would have to be a different set of separators from the ones with manual ordering. But this approach seems to be quite a pain in the neck, and I suggest just having separators only for manual sorting.

**Comment by decadent** [comment]
I second the request for separators, but implore the app authors to *not* implement them as additional horizontal lines. Dividing groups with simple empty space is much cleaner, whereas additional lines are just extra visual noise. There's some difficulty in that moving these empty separators might be somewhat non-obvious, but IMO people can adapt to that.

Here's my mockup for separators. (I also permitted myself to fiddle with the existing lines between individual items.)

![221](https://user-images.githubusercontent.com/240699/191989964-09af4f2b-9429-46f1-899c-2940288fb22f.png)
